mlx_graphs/data/batch.py

```diff
@@ -8,35 +7,38 @@
 
 class Batch(GraphData):
     """
-        Represents a batch data object describing a batch of graphs as one big (disconnected)
-        graph.
+    Represents a batch data object describing a batch of graphs as one big (disconnected)
+    graph.
 
     """
 
-    def __init__(self, node_features, edge_index, num_nodes) -> None:
+    def __init__(self, node_features, edge_index, cumsum) -> None:
         self.node_features = node_features
         self.edge_index = edge_index
-        self.cumsum_num_nodes = mx.cumsum(num_nodes)
+        self.cumsum = cumsum
 
     @classmethod
-    def from_graph_list(cls, graphs: List[GraphData]) -> Self :
+    def from_graph_list(cls, graphs: List[GraphData]) -> Self:
         """
         Constructs a :class:`mlx_graphs.batch.Batch` object from a
-        list of :class:`~mlx_graphs.data.GraphData` 
+        list of :class:`~mlx_graphs.data.GraphData`
         """
-        
+
         if not isinstance(graphs, (list, tuple)):
             graphs = list(graphs)
 
         num_nodes = mx.array([0] + [graph.num_nodes() for graph in graphs])
+        cumsum = mx.cumsum(num_nodes)
 
         global_node_features = [graph.node_features for graph in graphs]
         global_node_features = mx.concatenate(global_node_features, axis=0)
 
-        global_edge_index = [graph.edge_index + num_nodes[i] for i,graph in enumerate(graphs)]
+        global_edge_index = [
+            graph.edge_index + cumsum[i] for i, graph in enumerate(graphs)
+        ]
         global_edge_index = mx.concatenate(global_edge_index, axis=1)
 
-        return cls(global_node_features, global_edge_index, num_nodes = num_nodes)
+        return cls(global_node_features, global_edge_index, cumsum=cumsum)
 
     def __getitem__(self, idx):
```

Comment:
Have to write tests for this function please

Reply:
cls ? it's an internal function to construct an object out of the class the classmethod is part of, if you are referring to from_graph_list, the unit test, test it

Reply:
No __getitem__ ^^